Azure/osm-azure
# using global values for proxy override

This PR provides an example of using global values for proxy configuration. This strategy requires us to change the osm oss chart so that the proxy values live under the `global` key. See more info about Helm's global value [here](https://helm.sh/docs/topics/charts/#global-values). Global values allow the parent chart to share values with the subchart.

We'd also need to ask the extension team if they'd be okay with us scoping the proxy values under the global key.

From Helm's documentation:

```
This provides a way of sharing one top-level variable with all subcharts, which is useful for things like setting metadata properties like labels.

If a subchart declares a global variable, that global will be passed downward (to the subchart's subcharts), but not upward to the parent chart. There is no way for a subchart to influence the values of the parent chart.

Also, global variables of parent charts take precedence over the global variables from subcharts.
```

To test this, pull down this PR and run:
```$ helm template charts/osm-arc/ --set global.enableProxySupport=true > out.yaml```
Inspect out.yaml. You should see the proxy-config secret and the relevant proxy env vars in the osm-deployment template.

Then, run:
```$ helm template charts/osm-arc/ > out.yaml```
Inspect out.yaml. You should not see a template for proxy-config secret. You should also not see the proxy env vars in the osm-deployment template.


If this strategy works, we'd want to change `global.enableProxySupport` to `global.proxySettings.isProxyEnabled`, `global.httpProxy` to `global.proxySettings.httpProxy` and `global.httpsProxy` to `global.proxySettings.httpsProxy` as per the arc extension spec
